#### About

> Non parametric tests( e.g - Mann-Whitney U Test, Wilcoxon signed-rank test)                         

Non-parametric tests are statistical tests that do not make any assumptions about the underlying distribution of the data. These tests are used when the data does not follow a normal distribution or when the sample size is small. Two commonly used non-parametric tests are the Mann-Whitney U test and the Wilcoxon signed-rank test.

- Mann-Whitney U Test:
The Mann-Whitney U test is a non-parametric test used to compare two independent samples. It is also known as the Wilcoxon rank-sum test. The test is used to determine whether the two samples come from populations with the same distribution. The null hypothesis for the test is that the two populations have the same distribution.

- Use cases - In medical, It is used to compare the effectiveness of two treatments.

- Wilcoxon signed-rank test:
The Wilcoxon signed-rank test is a non-parametric test used to compare two related samples. It is used to determine whether the differences between the two samples are significant. The null hypothesis for the test is that there is no difference between the two populations.




Suppose we want to determine whether a new drug has a significant effect on blood pressure. We measure the blood pressure of 10 patients before and after administering the drug. We can use the Wilcoxon signed-rank test to determine whether there is a significant difference in the blood pressure measurements.



In [1]:
import numpy as np
from scipy.stats import wilcoxon

In [2]:
# Generate two related samples
before = np.random.normal(120, 10, 10)
after = np.random.normal(110, 10, 10)


In [3]:
# Apply Wilcoxon signed-rank test
stat, p = wilcoxon(before, after)
print('Wilcoxon signed-rank test')
print('Statistic:',stat)
print('p-value',p)


Wilcoxon signed-rank test
Statistic: 2.0
p-value 0.005859375


The obtained test statistic is 2.0 and the p-value is 0.005859375. Since the p-value is less than the commonly used significance level of 0.05, we reject the null hypothesis that there is no difference between the blood pressure measurements before and after the treatment. Therefore, we can conclude that the drug treatment has a significant effect on reducing blood pressure in patients.





> Using Mann-Whitney U Test to determine if there is a significant difference between the two groups

In [4]:
import numpy as np
from scipy.stats import mannwhitneyu

In [5]:
# Create two independent samples
group1 = [75, 82, 62, 68, 76, 74, 81, 68]
group2 = [55, 68, 72, 60, 70, 62, 65, 63]


In [6]:
# Perform Mann-Whitney U Test
stat, p = mannwhitneyu(group1, group2)


In [7]:
# Interpret results
alpha = 0.05
if p > alpha:
    print("Fail to reject null hypothesis - no significant difference between groups.")
else:
    print("Reject null hypothesis - significant difference between groups.")

Reject null hypothesis - significant difference between groups.
